### PR TITLE
AWS: add ACL support for S3FileIO

### PR DESCRIPTION
**Pull request by jackye1995**
As discussed in #1767 with @danielcweeks and @johnclara, add ACL support for `S3FileIO`

### CODE REVIEW

**Comment by johnclara** [comment]
lgtm!

**Comment by danielcweeks** [comment]
@jackye1995 and @johnclara we should probably figure out #1786 and make the ACL configuration follow a similar pattern.  It looks like we need to rebase on top of #1767 and add the configuration to the `CreateMultipartUploadRequest` as well.

**Comment by jackye1995** [comment]
> @jackye1995 and @johnclara we should probably figure out #1786 and make the ACL configuration follow a similar pattern. It looks like we need to rebase on top of #1767 and add the configuration to the `CreateMultipartUploadRequest` as well.

updated, following the same pattern as #1786 . I am intentionally separating them into `configureEncryption` and `configurePermission` to be explicit about the intention of the methods.

File: aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java [hunk 1]
```diff
@@ -165,6 +177,11 @@ public AwsProperties(Map<String, String> properties) {
 
     this.s3fileIoStagingDirectory = PropertyUtil.propertyAsString(properties, S3FILEIO_STAGING_DIRECTORY,
         System.getProperty("java.io.tmpdir"));
+
+    String aclType = properties.get(S3FILEIO_ACL);
+    this.s3FileIoAcl = ObjectCannedACL.fromValue(aclType);
+    Preconditions.checkArgument(s3FileIoAcl == null || !s3FileIoAcl.equals(ObjectCannedACL.UNKNOWN_TO_SDK_VERSION),
+        "Cannot supported S3 CannedACL " + aclType);
```
Comment (danielcweeks) [inline aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java:184]: nit: did you mean "Unsupported S3 CannedACL"?

**Comment by danielcweeks** [comment]
@jackye1995 One very minor nit, but other than that, this looks good to me.  If you could just update the message, I'll push it.

**Comment by jackye1995** [comment]
> @jackye1995 One very minor nit, but other than that, this looks good to me. If you could just update the message, I'll push it.

Thank you! Just fixed #1786 , I think there will be some merge conflicts with this one, if you can first merge that PR, I can rebase this one with the fix to the error message.

**Comment by danielcweeks** [comment]
> > @jackye1995 One very minor nit, but other than that, this looks good to me. If you could just update the message, I'll push it.
> 
> Thank you! Just fixed #1786 , I think there will be some merge conflicts with this one, if you can first merge that PR, I can rebase this one with the fix to the error message.

Both look good.  Just waiting on checks for #1786 and I'll push that one.

**Comment by jackye1995** [comment]
> > > @jackye1995 One very minor nit, but other than that, this looks good to me. If you could just update the message, I'll push it.
> > 
> > 
> > Thank you! Just fixed #1786 , I think there will be some merge conflicts with this one, if you can first merge that PR, I can rebase this one with the fix to the error message.
> 
> Both look good. Just waiting on checks for #1786 and I'll push that one.

fixed

**Pull request merged**